Sources/SwiftFormat/Rules/OmitReturns.swift

```diff
@@ -0,0 +1,157 @@
+///         equivalent `func <name>() { ... }` constructs.
+@_spi(Rules)
+public final class OmitReturns: SyntaxFormatRule {
+  public override class var isOptIn: Bool { return true }
+
+  public override func visit(_ node: FunctionDeclSyntax) -> DeclSyntax {
+    let decl = super.visit(node)
+
+    // func <name>() -> <Type> { return ... }
+    if var funcDecl = decl.as(FunctionDeclSyntax.self),
+       let body = funcDecl.body,
+       let `return` = containsSingleReturn(body.statements) {
+      funcDecl.body?.statements = unwrapReturnStmt(`return`)
+      diagnose(.omitReturnStatement, on: `return`, severity: .refactoring)
+      return DeclSyntax(funcDecl)
+    }
+
+    return decl
+  }
+
+  public override func visit(_ node: SubscriptDeclSyntax) -> DeclSyntax {
+    let decl = super.visit(node)
+
+    guard var `subscript` = decl.as(SubscriptDeclSyntax.self) else {
+      return decl
+    }
+
+    if let accessorBlock = `subscript`.accessorBlock,
+       // We are assuming valid Swift code here where only
+       // one `get { ... }` is allowed.
+       let transformed = transformAccessorBlock(accessorBlock) {
+      `subscript`.accessorBlock = transformed
+      return DeclSyntax(`subscript`)
+    }
+
+    return decl
+  }
+
+  public override func visit(_ node: PatternBindingSyntax) -> PatternBindingSyntax {
+    var binding = node
+
+    if let accessorBlock = binding.accessorBlock,
+       let transformed = transformAccessorBlock(accessorBlock) {
+      binding.accessorBlock = transformed
+      return binding
+    }
+
+    return node
+  }
+
+  public override func visit(_ node: ClosureExprSyntax) -> ExprSyntax {
+    let expr = super.visit(node)
+
+    // test { return ... }
+    if var closure = expr.as(ClosureExprSyntax.self),
+       let `return` = containsSingleReturn(closure.statements) {
+       closure.statements = unwrapReturnStmt(`return`)
+       diagnose(.omitReturnStatement, on: `return`, severity: .refactoring)
+       return ExprSyntax(closure)
+    }
+
+    return expr
+  }
+
+  private func transformAccessorBlock(_ accessorBlock: AccessorBlockSyntax) -> AccessorBlockSyntax? {
+    // We are assuming valid Swift code here where only
+    // one `get { ... }` is allowed.
+    switch accessorBlock.accessors {
+    case .accessors(let accessors):
+      guard var getter = accessors.filter({
+        $0.accessorSpecifier.tokenKind == .keyword(.get)
+      }).first else {
+        return nil
+      }
+
+      guard let body = getter.body,
+            let `return` = containsSingleReturn(body.statements) else {
+        return nil
+      }
+
+      guard let getterAt = accessors.firstIndex(where: {
+        $0.accessorSpecifier.tokenKind == .keyword(.get)
+      }) else {
+        return nil
+      }
+
+      getter.body?.statements = unwrapReturnStmt(`return`)
+
+      diagnose(.omitReturnStatement, on: `return`, severity: .refactoring)
+
+      return .init(
```

Comment:
At some point soon I'll be working on a modernization pass of the code base that starts doing regular mutation on nodes. Can you change this to just mutate a copy of the node and return that? For example,

```
var newBlock = accessorBlock
accessorBlock.accessors = .accessors(accessors.with(\.[getterAt], getter)
return newBlock
```

Likewise for the other branch below. This makes it easier to scan quickly and see what part of the node is being changed and what parts are staying the same.